### DISCUSSION TITLE
spot removal not being applied when create target image

### DISCUSSION THREAD

**Discussion by alanminner** [arkypita/LaserGRBL#1026] [Q&A]
I am trying to set up engraving using vector but the spot removal setting is not being applied.  I am using version v3.3.1

**Comment by gmmanonymus111** [comment]
+1
LaserGRBL shows the preview right, but does not actually remove anything.
![image](https://user-images.githubusercontent.com/9658243/83667792-430d9b80-a5cf-11ea-97a1-0fc5c57855fa.png)
![image](https://user-images.githubusercontent.com/9658243/83667825-4acd4000-a5cf-11ea-98fa-de26b337317c.png)

**Comment by arkypita** [comment]
Hi all
There is one thing to know when using raster import.
All the parameters taken have an effect on a preview image which is usually much smaller than that which is usually processed in the final phase. This is for performance issues in the preview generation.

This causes that the result could be very different from what you will see in the final image.

In a shrunken image, defects and "spots" are also shrunk, so a low threshold can easily remove them. When the image is instead treated at full resolution (or even sized up) the same threshold has no effect because the spots are much larger.

**Comment by arkypita** [comment]
You can also see effect of this if you maximize/reduce the size of raster import form.

This is difficult to change, even accepting a longer preview design time, because the final size the user wants is not known in advance.

a solution could be to always apply vectorization to the original image (neither enlarged, nor reduced) so as to have a greater reproducibility of the result, both in the preview phase and in the marking phase on different sizes.

However, this changes may not be entirely appreciated as it is an interesting thing that the drawing is "simplified" and "cleaned" automatically when printed in small size, and that it is enriched with details when reproduced in large dimensions.
